monai/apps/datasets.py

```diff
@@ -278,17 +280,18 @@ def get_properties(self, keys: Optional[Union[Sequence[str], str]] = None):
     def _generate_data_list(self, dataset_dir: str) -> List[Dict]:
         section = "training" if self.section in ["training", "validation"] else "test"
         datalist = load_decathlon_datalist(os.path.join(dataset_dir, "dataset.json"), True, section)
-        if section == "test":
+        return self._split_datalist(datalist)
+
+    def _split_datalist(self, datalist: List[Dict]) -> List[Dict]:
```

Comment:
could you add more tests?

Reply:
Sure, added more unit tests in the latest commit.
Thanks.